packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/diagramfilter/services/DiagramFilterHelper.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.views.diagramfilter.services;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import org.eclipse.sirius.components.collaborative.api.ChangeDescription;
+import org.eclipse.sirius.components.collaborative.api.IEditingContextEventProcessor;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationEventProcessor;
+import org.eclipse.sirius.components.collaborative.diagrams.api.IDiagramInput;
+import org.eclipse.sirius.components.core.api.ErrorPayload;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.core.api.SuccessPayload;
+import org.eclipse.sirius.components.representations.Failure;
+import org.eclipse.sirius.components.representations.IStatus;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import reactor.core.publisher.Sinks;
+import reactor.core.publisher.Sinks.EmitResult;
+import reactor.core.publisher.Sinks.Many;
+import reactor.core.publisher.Sinks.One;
+
+/**
+ * Utility methods for the "Diagram Filter" view.
+ *
+ * @author gdaniel
+ */
+@Service
+public class DiagramFilterHelper {
+
+    private final Logger logger = LoggerFactory.getLogger(DiagramFilterHelper.class);
+
+    public Set<String> getSelectedElementIds(VariableManager variableManager) {
+        Map<String, Boolean> checkMap = variableManager.get(DiagramFilterDescriptionProvider.SELECTED_TREE_NODES, Map.class).get();
+        return checkMap.entrySet().stream()
+                .filter(e -> e.getValue().equals(Boolean.TRUE))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet());
+    }
+
+    public IStatus sendDiagramEvent(VariableManager variableManager, IDiagramInput diagramInput) {
+        IRepresentationEventProcessor representationEventProcessor = variableManager.get("diagramEventProcessor", IRepresentationEventProcessor.class).get();
+        IEditingContextEventProcessor editingContextEventProcessor = variableManager.get("editingContextEventProcessor", IEditingContextEventProcessor.class).get();
```

Comment:
Don't use `.get()` on optional without testing that the value is present first.

Reply:
done